docs/projects/selection-sort/_posts/2019-01-10-python.md

```diff
@@ -0,0 +1,176 @@
+
+## How to Implement the Solution
+
+At this point, let's dig into the code a bit. The following sections break
+down the Selection Sort in Python functionality.
+
+### Solution
+
+```python
+#!/usr/bin/env python
+import sys
+
+
+def selection_sort(xs, sorted_xs=None):
+    sorted_xs = sorted_xs or []
+    if len(xs) <= 0:
+        return sorted_xs
+    x = min(xs)
+    sorted_xs.append(x)
+    xs.remove(x)
+    return selection_sort(xs, sorted_xs)
+
+
+def input_list(list_str):
+    return [int(x.strip(" "), 10) for x in list_str.split(',')]
+
+
+def exit_with_error():
+    print('Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”')
+    sys.exit(1)
+
+
+def main(args):
+    try:
+        xs = input_list(args[0])
+        if len(xs) <= 1:
+            exit_with_error()
+        print(selection_sort(xs))
+    except (IndexError,ValueError):
+        exit_with_error()
+
+
+if __name__ == "__main__":
+    main(sys.argv[1:])
+```
+
+### main()
+
+Breaking down this solution bottom up,
+
+```python
+if __name__ == "__main__":
+    main(sys.argv[1:])
+```
+
+This bit of code checks to see if this is the main module run. If it is it then calls the main function and passes user input to it.
+In this case the user input would be a string of numbers like so `"2, 1, 10, 5, 3"` (to sort).
+
+```python
+def main(args):
+    try:
+        xs = input_list(args[0])
+        if len(xs) <= 1:
+            exit_with_error()
+        print(selection_sort(xs))
+    except (IndexError,ValueError):
+        exit_with_error()
+```
+
+This is the main function of this file. It parses the input, then calls our selection sort function (and prints the results).
+It also deals with any errors raised.
+
+### input_list()
+
+```python
+def input_list(list_str):
+    return [int(x.strip(" "), 10) for x in list_str.split(',')]
+```
+
+This function takes a string like `"2, 1, 10, 5, 3"`, and turns into a list of numbers. It does this using a list comprehension, first we need to convert
+our string into a list `list_str.split(',')` which is a list of strings split by comma (,). So our original input string becomes `["2", " 1", " 10", " 5", " 3"]`.
+Then for each element in the list `for x in ...` ,  we do something to it.
+
+In this example we convert it into a decimal integer, `int(x.strip(" "), 10)`
+`x.strip(" ")`, removes any whitespace so `" 1"` becomes `"1"`, then `int("1", 10)` converts the string `"1"` into a decimal number in this case very simply `1`. This is done
+for every item in the list so our original input of `"2, 1, 10, 5, 3"` becomes `[2, 1, 10, 5, 3]`.
+
+### exit_with_error()
+
+```python
+def exit_with_error():
+    print('Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”')
+    sys.exit(1)
+```
+
+This function simply prints a message and then exits the script with an error, `sys.exit(1)`. If any non-zero value is returned then the program didn't complete properly.
```

Comment:
I did two full reads of your solution, and I saw the word "simply" sprinkled throughout. Often, you'll find that I use the word as well, but it's lately been considered bad practice because it can seem a little condescending. I **know** that's not your intent as I use the word by accident sometimes as well, but I wanted to point it out.

Reply:
Yeh thats makes sense. I'll look at rewording these.